numba/tests/test_lists.py

```diff
@@ -1770,7 +1770,7 @@ def foo(x):
 
         with self.assertRaises(errors.TypingError) as raises:
             foo(100)
-        expect = "Cannot unify LiteralList"
+        expect = "Cannot infer"
```

Comment:
I'm not sure these changes are quite right? I think this is a unification problem, not an inference problem, the types of the `l` variables can be inferred, however `l` cannot be unified across the branch as there is no type which encompasses the version if true and the version if false.